interactions/client/bot.py

```diff
@@ -339,6 +237,167 @@ async def wait_until_ready(self) -> None:
         """Helper method that waits until the websocket is ready."""
         await self._websocket.wait_until_ready()
 
+    async def __get_all_commands(self) -> None:
+        # this method is just copied from the sync method
+        # I expect this to be changed in the sync rework
+        # until then this will deliver a cache if sync is off to make autocomplete work bug-free
+        # but even with sync off, we should cache all commands here always
+
+        _guilds = await self._http.get_self_guilds()
+        _guild_ids = [int(_["id"]) for _ in _guilds]
+        self._scopes.add(_guild_ids)
+        _cmds = await self._http.get_application_commands(
+            application_id=self.me.id, with_localizations=True
+        )
+
+        for command in _cmds:
+            if command.get("code"):
+                # Error exists.
+                raise JSONException(command["code"], message=f'{command["message"]} |')
+
+        self.__global_commands = {"commands": _cmds, "clean": True}
+        # TODO: add to cache (later)
+
+        # responsible for checking if a command is in the cache but not a coro -> allowing removal
+
+        for _id in _guild_ids:
+            _cmds = await self._http.get_application_commands(
+                application_id=self.me.id, guild_id=_id, with_localizations=True
+            )
+
+            for command in _cmds:
+                if command.get("code"):
+                    # Error exists.
+                    raise JSONException(command["code"], message=f'{command["message"]} |')
+
+            self.__guild_commands[_id] = {"commands": _cmds, "clean": True}
+
+    async def __sync(self) -> None:  # sourcery no-metrics
+        """Synchronizes all commands to the API."""
+
+        log.debug("starting command sync")
+        _guilds = await self._http.get_self_guilds()
+        _guild_ids = [int(_["id"]) for _ in _guilds]
+        self._scopes.add(_guild_ids)
```

Comment:
Can't hash a list.

Reply:
How can I put the items from the list into the set? Is there like a method or do I have to do a for-loop?

Reply:
Instead of `add()`, switch it to `update()`

Reply:
With the same argument?

Reply:
Yes.

Reply:
Cool, thanks

Reply:
Done